src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -102,31 +105,22 @@ internal class GrpcWorkerChannel : IRpcWorkerChannel, IDisposable
             _applicationHostOptions = applicationHostOptions;
             _sharedMemoryManager = sharedMemoryManager;
             _workerConcurrencyOptions = workerConcurrencyOptions;
+            _processInbound = state => ProcessItem((InboundGrpcEvent)state);
 
             _workerCapabilities = new GrpcCapabilities(_workerChannelLogger);
 
-            _inboundWorkerEvents = _eventManager.OfType<InboundGrpcEvent>()
-                .Where(msg => msg.WorkerId == _workerId);
-
-            _eventSubscriptions.Add(_inboundWorkerEvents
-                .Where(msg => msg.IsMessageOfType(MsgType.RpcLog) && !msg.IsLogOfCategory(RpcLogCategory.System))
-                .Subscribe(Log));
-
-            _eventSubscriptions.Add(_inboundWorkerEvents
-                .Where(msg => msg.IsMessageOfType(MsgType.RpcLog) && msg.IsLogOfCategory(RpcLogCategory.System))
-                .Subscribe(SystemLog));
+            if (_eventManager.TryGetGrpcChannels(workerId, out var inbound, out var outbound))
+            {
+                _outbound = outbound.Writer;
+                _inbound = inbound.Reader;
+                // note: we don't start the read loop until StartWorkerProcessAsync is called
+            }
```

Comment:
The factory should create the channels prior to creating the `GrpcWorkerChannel`, so we should always have these channels setup here. Are expecting this to ever return `false`? Should we throw that is ever the case?